### PR TITLE
Improve javadoc for Hazelcast 3 connector HZ-373

### PR DESCRIPTION
**Pull request by frant-hartm**
Add top level javadoc for source and sink
Mark as beta

HZ-373

### CODE REVIEW

File: extensions/hazelcast-3-connector/hazelcast-3-connector-common/src/main/java/com/hazelcast/connector/Hz3Enrichment.java [hunk 1]
```diff
@@ -22,48 +22,57 @@
...
- * the map, a {@link ServiceFactory} to Hazelcast 3 Map is used. This class provides a utility
- * methods to create and use this service factory.
+ * Provides a way to perform enrichment using a Map or ReplicatedMap from
+ * Hazelcast 3 cluster.
+ * <p>
+ * The usage is similar to
+ * {@link GeneralStage#mapUsingIMap(String, FunctionEx, BiFunctionEx)} and
+ * other similarly named methods. The difference is that instead of providing
+ * a name of the map, a {@link ServiceFactory} to Hazelcast 3 Map is used.
+ * This class provides a utility methods to create and use this service factory.
```
Comment (Serdaro) [inline extensions/hazelcast-3-connector/hazelcast-3-connector-common/src/main/java/com/hazelcast/connector/Hz3Enrichment.java:40]: ... provides utility methods ...

**Review by Serdaro — APPROVED** [review]
A minor comment. Other than that, LGTM.

File: extensions/hazelcast-3-connector/hazelcast-3-connector-common/src/main/java/com/hazelcast/connector/Hz3Sinks.java [hunk 2]
```diff
@@ -16,21 +16,55 @@
...
+ * config.addCustomClasspaths(sink.name(), jars)
+ * }</pre>
+ * <p>
+ * The jars must exist in the directory specified by the
+ * {@link ClusterProperty#PROCESSOR_CUSTOM_LIB_DIR}
+ * directory. This is already set up for the regular zip distribution.
+ *
+ * @since 5.0
  */
+@Beta
```
Comment (ufukyilmaz) [inline extensions/hazelcast-3-connector/hazelcast-3-connector-common/src/main/java/com/hazelcast/connector/Hz3Sinks.java:67]: Beta annotation can also be added to the methods of this class.

File: extensions/hazelcast-3-connector/hazelcast-3-connector-common/src/main/java/com/hazelcast/connector/Hz3Sinks.java [hunk 3]
```diff
@@ -16,21 +16,55 @@
...
 import static com.hazelcast.jet.core.ProcessorMetaSupplier.preferLocalParallelismOne;
 
 /**
- * Sinks to connect to Hazelcast 3 cluster
+ * Provides sinks to write to Hazelcast 3 cluster.
+ * <p>
+ * The usage is similar to {@link Sinks#remoteMap(String, ClientConfig)}.
+ * Because of incompatible APIs the configuration is passed as an XML
+ * document in a string - note that the XML configuration must conform
+ * to 3.x schema.
```
Comment (ufukyilmaz) [inline extensions/hazelcast-3-connector/hazelcast-3-connector-common/src/main/java/com/hazelcast/connector/Hz3Sinks.java:40]: Here, maybe we can direct the user to: https://www.hazelcast.com/schema/config/. It's entirely up to you, mine is just as a optional suggestion (if added here it should be added to Hz3Sources as well).

**Review by ufukyilmaz — APPROVED** [review]
LGTM

**Review by ufukyilmaz — CHANGES_REQUESTED** [review]
it might be good to have a reference to the Hazelcast schema repository from the related javadocs

File: extensions/hazelcast-3-connector/hazelcast-3-connector-common/src/main/java/com/hazelcast/connector/Hz3Sources.java [hunk 4]
```diff
@@ -21,18 +21,50 @@
...
+ * } </pre>
+ * Additionally, a custom classpath element for the {@code source} stage
+ * must be set with the Hazelcast 3 client and the Hazelcast 3 connector:
+ * <pre>{@code
+ * List<String> jars = new ArrayList<>();
+ * jars.add("hazelcast-3.12.12.jar");
+ * jars.add("hazelcast-client-3.12.12.jar");
+ * jars.add("hazelcast-3-connector-impl.jar");
+ * JobConfig config = new JobConfig();
+ * config.addCustomClasspaths(mapStage.name(), jars)
```
Comment (ufukyilmaz) [inline extensions/hazelcast-3-connector/hazelcast-3-connector-common/src/main/java/com/hazelcast/connector/Hz3Sources.java:58]: minor `mapStage.name()` -> `source.name()`

File: extensions/hazelcast-3-connector/hazelcast-3-connector-common/src/main/java/com/hazelcast/connector/Hz3Enrichment.java [hunk 5]
```diff
@@ -22,48 +22,62 @@
...
- * client:
+ * And finally, a custom classpath element for the {@code mapUsingService}
+ * stage must be set with the Hazelcast 3 client:
  * <pre>{@code
  * List<String> jars = new ArrayList<>();
  * jars.add("hazelcast-3.12.12.jar");
  * jars.add("hazelcast-client-3.12.12.jar");
  * jars.add("hazelcast-3-connector-impl.jar");
- * config.addCustomClasspaths(name, jars);
+ * config.addCustomClasspaths(source.name(), jars);
```
Comment (ufukyilmaz) [inline extensions/hazelcast-3-connector/hazelcast-3-connector-common/src/main/java/com/hazelcast/connector/Hz3Enrichment.java:72]: minor: `source.name()` -> `mapStage.name()`

**Review by ufukyilmaz — APPROVED** [review]

**Pull request merged**